File: tests/fixtures/real_world/rust/taint/ssrf_url_only_no_data_exfil.rs
// Regression: a tainted URL flowing through a reqwest chain *without*
// a body-binding step must fire SSRF (taint-unsanitised-flow) but must
// NOT fire DATA_EXFIL.  The chain text reduces to `Client::new.post`
// with no `body|json|form|multipart` segment, so the body-bind chain
// matcher cannot attach.  Guards against the new chain-aware DATA_EXFIL
// rule over-firing on pure URL flows.
fn fetch_url_only() {
    let url = std::env::var("TARGET_URL").unwrap();
    let _ = reqwest::Client::new().post(&url).send();
}

File: tests/fixtures/java/ssrf_url_only_no_data_exfil.java
// Regression fixture: a tainted URL flowing into HttpClient.send must
// fire SSRF (taint-unsanitised-flow) but must NOT fire DATA_EXFIL.
// The body is a hardcoded literal so no Sensitive payload reaches the
// outbound request.  This guards against over-firing DATA_EXFIL on
// flows where only the URL position is attacker-controlled.
//
// Driven by `data_exfil_java_integration_tests.rs`.
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpRequest.BodyPublishers;
import java.net.http.HttpResponse.BodyHandlers;
import javax.servlet.http.HttpServletRequest;

public class SsrfUrlOnlyNoDataExfil {
    public void doGet(HttpServletRequest request) throws Exception {
        String target = request.getParameter("url");
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest req = HttpRequest.newBuilder()
            .uri(URI.create(target))
            .POST(BodyPublishers.ofString("ping"))
            .build();
        client.send(req, BodyHandlers.ofString());
    }
}
